src/dialects/abstract/data-types.ts

```diff
@@ -1870,20 +1870,20 @@ export class ENUM<Member extends string> extends AbstractDataType<Member> {
    * @param options either array of values or options object with values array. It also supports variadic values.
    */
   constructor(options: EnumOptions<Member>);
-  constructor(members: Member[]);
-  constructor(...members: Member[]);
+  constructor(members: Member[] | readonly Member[]);
+  constructor(...members: Member[] | readonly Member[]);
   // we have to define the constructor overloads using tuples due to a TypeScript limitation
   //  https://github.com/microsoft/TypeScript/issues/29732, to play nice with classToInvokable.
   /** @internal */
   constructor(...args:
     | [options: EnumOptions<Member>]
-    | [members: Member[]]
-    | [...members: Member[]]
+    | [members: Member[] | readonly Member[]]
+    | [...members: Member[] | readonly Member[]]
   );
   constructor(...args: [Member[] | Member | EnumOptions<Member>, ...Member[]]) {
     super();
 
-    let values: Member[];
+    let values: Member[] | readonly Member[];
```

Comment:
once assigned to this.options, `values` should only be a readonly array. We need a `NormalizedEnumOptions` type which only accepts readonly arrays